### DISCUSSION TITLE
Remove ROS key from docker image

### DISCUSSION THREAD

**Discussion by franklinselva** [space-ros/space-ros#281] [Ideas]
The current builds of space ROS docker image have ROS2 apt keys stacked up for handling dependencies. The keys are left as a part of the final image which allows us to install an `apt` version of the ros packages. From discussion #188, it would be nice to remove these keys from the final image making anyone prefer source builds for the projects.

The fix for this issue and the subsequent stacked variant of images would be,

1. Remove ros2 apt-key from the **base**, **core** and **dev** version of the images.
2. For **Nav2** and **Moveit2** variants, we can add the keys temporarily for managing dependencies and remove them by the end.
3. For **demo** variants, prefer source builds instead of using ROS2 apt keys.

**Comment by asimonov** [comment]
i think it is a great idea to prevent installing ros2-xxx-pkgxxx. but do ros2 setup tools come from same apt repo? what would happen with manual apt update/install in the container if we do this?

**Comment by franklinselva** [comment]
@asimonov The ros2 setup tools are a part of the `dev` image variant which comes pre-installed. This shouldn't be a problem with the proposal.  

However, the user can still manually set up the apt key and proceed with the installation at their own risk (for instance, quick testing of new algorithms before proceeding with the guidelines). This will deviate from the strict guidelines the space-ros project is trying to impose but allowing this flexibility could benefit many areas of development at its initial phase.
